pelican/utils.py

```diff
@@ -603,6 +603,25 @@ def truncate_html_words(s, num, end_text='…'):
     return out
 
 
+def truncate_html_paragraphs(s, count):
+    """Truncates HTML to a certain number of paragraphs.
+
+    :param count: number of paragraphs to keep
+
+    Newlines in the HTML are preserved.
+    """
+    paragraphs = []
+    tag_stop = 0
+    substr = s[:]
+    for i in range(count):
+        substr = substr[tag_stop:]
+        tag_start = substr.find('<p>')
+        tag_stop = substr.find('</p>') + len('</p>')
```

Comment:
My time on the interwebs taught me that the possibility of finding a closing tag for `<p>` is like a coin toss. Even the HTML spec is permissive about it and "formally" allows omitting `</p>` in most cases. Long story short, I am trying to say that this is quite unreliable.

Reply:
The HTML is mostly generated from a generator (reST, Markdown…) that hopefully emits closing tags, so in practice this should not happen very much.

Reply:
> in practice this should not happen very much.

That doesn't cut it though :). `rst` and `md` generators may play nice (for now) but content can come from any number for sources not limited to those, including direct `html` files, plugins for other formats etc. In software, edge cases happen more common than one would think initially :).

I'm not suggesting employing full DOM parsers like `lxml` or similar, although that would be more complete. But that can be postponed until the need arises. It should at least produce something reasonable in those cases: i.e. if `</p>` is not present (`substr.find('</p>') == -1`), take up to the next `<p>`?